whois-commons/src/main/java/net/ripe/db/whois/common/dao/ResourceDataDao.java

```diff
@@ -116,5 +116,25 @@ public int compareTo(final State other) {
                 }
             }
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            State state = (State) o;
+
+            if (id != state.id) return false;
```

Comment:
these three statements can be folded into one: return (id == state.id) && (count == state.count) && ...